nursery/encrypt-data-using-rc4-via-systemfunction032.yml

```diff
@@ -0,0 +1,26 @@
+rule:
+  meta:
+    name: encrypt data using RC4 via SystemFunction032
+    namespace: data-manipulation/encryption/rc4
+    authors:
+      - richard.weiss@mandiant.com
+    scope: function
+    att&ck:
+      - Defense Evasion::Obfuscated Files or Information [T1027]
+    mbc:
+      - Defense Evasion::Obfuscated Files or Information::Encryption-Standard Algorithm [E1027.m05]
+      - Cryptography::Encrypt Data::RC4 [C0027.009]
+    references:
+      - https://doxygen.reactos.org/df/d13/sysfunc_8c_source.html
+      - https://blog.gentilkiwi.com/tag/systemfunction032
+    examples:
+      - 3BBDF04C25FCD9876733EAA9163B3ED64D81396E7414619758D9376EDF4C103E:0x1000976C # api match
+  features:
+    - or:
+      - api: SystemFunction032
+      - and:
+        - match: link function at runtime on Windows
+        - string: "SystemFunction032"
+        - or:
```

Comment:
```suggestion
        - optional:
```

Reply:
or was intended to use is as 1 or mroe (at least 1) to avoid that this is a string only match before I added "matching of link function at runtime". The decision of at least 1 and not 0 or more was inteded to have a more robust rule. With match we could think about chaning it to:
```
- and:
- string: "systemfunction032"
- or:
- -string: /..../i  
- -string: /.../i
- -match: /../i
```
opportunities:
- optional (0 or more) + AND (your recommendation )-> at least 1 -> or (rule change above) perhaps with performance relevance
- optional (0 or more) + AND for better reading (your recommendation )
- AND + OR for stronger robustness of the rule to avoid using "SystemFunction032" as sting as decoy and a indirect/resolved calling of another function

My recommendation:
Would stay with the last option and loking from time to time in VT the differences between content match of only Systemfunction032 and combined with the module names.

Reply:
can you elaborate on the robustness of the last option? I think with the string and match present we can be pretty sure (of course not absolutely) what's happening... do the additional DLL strings add more value vs. potential misses where they are missing

Reply:
friendly ping @richardweiss80